### DISCUSSION TITLE
Purolator Rate Estimate / Shopping -- 'Full' vs 'Quick'?

### DISCUSSION THREAD

**Discussion by ChrisNolan** [karrioapi/karrio#741] [Q&A]
It looks like the system is using the 'GetFullEstimate' instead of the GetQuickEstimate API call to purolator... so when I try to just get postal code to postal code it is failing with an error about City.  

What would be a good way to go about allowing both to work (my brute for for our fork might be to just switch it over to the other but...)?  I see that the code is there in 'estimate_service_2_1_2.py' but isn't imported in 'rate.py' for purolator.

Do the other carriers of similar different types of ratings that can be called?

**Comment by danh91** [comment]
Hi @ChrisNolan ,

This is good I don't remember off the top of my mind, but there was a limitation in terms of data returned from the `GetQuickEstimate` 🤔  the first implementation was using that. 

looking into it.

**Comment by ChrisNolan** [comment]
Thanks.  Turns out, once I looked at the backend which I was replacing, it turns out I have the full addresses available at that point anyway (and they were being sent to the Canpar call as it was just not the purolator which I was doing first) so I'm fine using the `GetFullEstimate` anyway.  And so much nicer to doing a single call and getting them all rather than multiple separate calls 🎉
